src/api/app/controllers/trigger_workflow_controller.rb

```diff
@@ -34,6 +34,10 @@ def event
     @github_event || @gitlab_event
   end
 
+  def github_enterprise_host
+    request.headers['X-GitHub-Enterprise-Host']
```

Comment:
There is no way to get the URL from the webhook payload? Because with this implementation, it's creeping up in many different classes...

Reply:
The payloads are very similar in both cases (enterprise/public). Only headers contain this information.

Reply:
I had a look at GitHub's documentation on webhook events and payloads and just as an example for the `push` event, there are multiple fields with the URL in the payload: https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push.

Same story for the `pull_request` event. It seems to be the case for all events.

Reply:
I see. Okay, I'm updating it.

Reply:
@dmarcoux how can we differentiate the requests from enterprise and public instances? It's important to differentiate the requests because later on, we use `octokit` and octokit requires api_endpoint for enterprise API calls.
https://github.com/octokit/octokit.rb#interacting-with-the-githubcom-apis-in-github-enterprise

We need a way to differentiate these requests, this is what we do by checking the header `X-GitHub-Enterprise-Host`. We can also do that by having a regex for URLs coming from payload but that would not be a solid solution. What do you think?

Reply:
> the requests because later on, we use octokit and octokit requires api_endpoint for enterprise API calls.

Why do we need to?

Reply:
@dmarcoux its updated. I’ve removed the additional parameter.

Reply:
@hennevogel I know atm we only support public repos but it was easy to handle endpoints for private instances in a single go. At this point we just have the ability to separate public and private repos. In future if we decide to add support for private instances then we already have the implementation here.